### PR TITLE
feat(examples/templates): add GCP VM devcontainer template

### PR DESCRIPTION
**Pull request by matifali**
Part of #10735

### CODE REVIEW

File: examples/templates/gcp-devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+display_name: Google Compute Engine (Devcontainer)
+description: Provision a Devcontainer on Google Compute Engine instances as Coder workspaces
+icon: ../../../site/static/icon/gcp.png
+maintainer_github: coder
+verified: true
+tags: [vm, linux, gcp, devcontainer]
+---
+
+# Remote Development in a Devcontainer on Google Compute Engine
```
Comment (bpmct) [inline examples/templates/gcp-devcontainer/README.md:10]: Thoughts on making a basic architecture diagram that shows that there is a VM that runs the docker container? I know that is a bit of extra work but could help make it clear.

Just looked up the [Codespaces architecture](https://docs.github.com/en/codespaces/overview) and they do that.

File: examples/templates/gcp-devcontainer/main.tf [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+data "google_compute_default_service_account" "default" {
+}
+
+data "coder_workspace" "me" {
+}
+
+resource "google_compute_disk" "root" {
+  name  = "coder-${data.coder_workspace.me.id}-root"
+  type  = "pd-ssd"
+  image = "debian-cloud/debian-12"
```
Comment (bpmct) [inline examples/templates/gcp-devcontainer/main.tf:68]: Could we use an image that includes docker pre-installed?

Comment (matifali) [reply examples/templates/gcp-devcontainer/main.tf:68]: I looked into it and an initial search didn't find any official/trust worry images. Most of the images are 3rd party, outdated and charge for usage charges too.

File: examples/templates/gcp-devcontainer/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+   during the setup process.
+
+## Architecture
+
+This template provisions the following resources:
+
+- GCP VM (persistent)
+- GCP Disk (persistent, mounted to root)
+
+Coder persists the root volume. The full filesystem is preserved when the workspace restarts. See this [community example](https://github.com/bpmct/coder-templates/tree/main/aws-linux-ephemeral) of an ephemeral AWS instance.
```
Comment (bpmct) [inline examples/templates/gcp-devcontainer/README.md:57]: This links to AWS not GCP

Comment (matifali) [reply examples/templates/gcp-devcontainer/README.md:57]: I thought it's intentional it's also present in GCP VM template

Comment (matifali) [reply examples/templates/gcp-devcontainer/README.md:57]: If it's a mistake I will update that too I'm a separate PR.

File: examples/templates/gcp-devcontainer/main.tf [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+      echo "Docker not found, installing..."
+      curl -fsSL https://get.docker.com -o get-docker.sh && sudo sh get-docker.sh 2>&1 >/dev/null
+      sudo usermod -aG docker ${local.linux_user}
+      newgrp docker
+    else
+      echo "Docker is already installed."
+    fi
+    # Start envbuilder
+    docker run --rm \
+      -v /tmp/envbuilder:/workspaces \
```
Comment (bpmct) [inline examples/templates/gcp-devcontainer/main.tf:168]: I think this one is good now too as soon as you move this to the home dir

**Review by bpmct — APPROVED** [review]

File: examples/templates/gcp-devcontainer/main.tf (hunk 4 above)
Comment (matifali) [reply examples/templates/gcp-devcontainer/main.tf:168]: I am holding it to be sure it works. For my final testing I am not able to get the agent connected

**Comment by matifali** [comment]
@ericpaulsen @bpmct FYI, I am also adding both `aws-devcontainer` and `gcp-devcontainer` to starter templates.

**Pull request merged**